lib/syntax_tree/cli.rb

```diff
@@ -239,6 +239,20 @@ class << self
       def run(argv)
         name, *arguments = argv
 
+        # If there are any plugins specified on the command line, then load them
+        # by requiring them here. We do this by transforming something like
+        #
+        #     stree format --plugins=haml template.haml
+        #
+        # into
+        #
+        #     require "syntax_tree/haml"
+        #
+        if arguments.first&.start_with?("--plugins=")
+          plugins = arguments.shift[/^--plugins=(.*)$/, 1]
+          plugins.split(",").each { |plugin| require "syntax_tree/#{plugin}" }
```

Comment:
Doing a self review, it strikes me that it's not obvious how one would register a third-party plugin (that isn't part of the `syntax_tree` gem).

I suppose another gem could expose a `lib/syntax_tree/foo` source, which might work in a Bundler setting (where the gem is pre-activated and its `lib` is added to the load paths) ... when using system gems, I'm not sure this would work, as I vaguely recall some magic with `require`'ing a gem name and auto-activating it... it's been too long since I've been without Bundler!

Anyhow: my comment applies to all uses of this option, `lsp` or no; not particularly relevant to this PR.

Reply:
I tried to be explicit about this in the README (https://github.com/ruby-syntax-tree/syntax_tree#configuration). It should be fine to add a `syntax_tree` directory under a path that is already in the require path. There are a lot of gems that do this (for example any Rails plugin that adds generators).

If folks need other gems activated that they for some reason can't require in their own plugins, I'm going to say they should use `RUBYOPT` instead. This isn't meant to be a whole replacement for the require system, just a small convenience option.

Reply:
I hadn't considered `RUBYOPT` that ought to be sufficient for anyone's needs.